City-of-Helsinki/drupal-hdbt
# UHF-2929 embedded content cookie

**How to test:**
* Run `composer require 'drupal/hdbt:dev-UHF-2929-embedded-content-cookie'` to get the changes.
* Run `composer require  'drupal/helfi_platform_config:dev-UHF-2929-embedded-content-cookie-translations'` to get the translation changes.
* Run `make drush-locale-update` to import the new translations.
* Run `drush cr` to clear the caches.
* Create a page that has one or many remote YouTube videos.
* Check that for a user who has _preference_ and _statistics_ cookie categories enabled, the video is loaded as an embedded iframe.
* Check that for a user who _do not_ have _preference_ and _statistics_ cookie categories enabled, the video is not loaded as embedded iframe. Instead, the user sees a message which gives information why the video is not loaded and how to watch the video.
  * Check that the message corresponds the design.
  * You could also check that there's no YouTube related cookies and local storage items present before user has approved the relevant cookie categories.
* Review the code, especially the parts related to building the iframe.

Also check the PR with the translations: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/209.
